alica_engine/include/engine/parser/PlanParser.h

```diff
@@ -37,22 +37,25 @@ class PlanParser
     void ignoreMasterPlanId(bool val);
     std::map<int64_t, AlicaElement*>* getParsedElements();
 
-    const std::string& getCurrentFile() const { return currentFile; }
+    const std::string& getCurrentFile() const { return _currentFile; }
     void setCurrentFile(const std::string& currentFile);
     void parseFileLoop();
     const RoleSet* parseRoleSet(std::string roleSetName);
     int64_t parserId(tinyxml2::XMLElement* node);
 
 private:
-    std::shared_ptr<ModelFactory> mf;
-    PlanRepository* rep;
-    Plan* masterPlan;
-    std::string planDir;
-    std::string roleDir;
-    std::string basePlanPath;
-    std::string baseRolePath;
-    std::string currentDirectory;
-    std::string currentFile;
+    std::shared_ptr<ModelFactory> _mf;
+    PlanRepository* _rep;
+    Plan* _masterPlan;
+    std::string _planDir;
+    std::string _roleDir;
+    std::string _basePlanPath;
+    std::string _baseRolePath;
+    std::string _currentDirectory;
+    std::string _currentFile;
+    std::list<std::string> _filesToParse;
```

Comment:
vectors

Reply:
parsefileinloop seems to be recursively modifying this. using vector in an efficient way would change the order of parsing files. Not sure whether it is acceptable.